### PR TITLE
fix: improve verify command s.t. os envs not passed through to container envs and instead add a flag for this purpose

### LINKED ISSUE #7945: Skaffold verify overrides container's env with host's env
Skaffold verify overrides container's env with host's env and fails to run container's commands due to invalid PATH, GOPATH and other environment variables.

Details:

Dockerfile
```
FROM golang:1.18
WORKDIR /src
COPY . .
CMD ["go", "test", "./..."]
```

Error
```
1 error(s) occurred:
* creating container in local docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "go": executable file not found in $PATH: unknown
```

`docker inspect <container>` if run via `skaffold verify -a build_artifacts.json`
```javascript
[
    {
        // ...
        "Config": {
            // ...
            "StdinOnce": false,
            "Env": [
                "PATH=/opt/homebrew/bin:/Users/<redacted>/.go/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin",
                "SHELL=/bin/fish",
                "GOPATH=/Users/<redacted>/.go"
                ...
            ]
            // ...
        }
        // ...
    }
]
```
It clearly contains PATH and other environment variables from my host machine.

`docker inspect <container>` if run via `docker run`:
```javascript
[
    {
        // ...
        "Config": {
            // ...
            "StdinOnce": false,
            "Env": [
                "PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "GOPATH=/go"
                ...
            ]
            // ...
        }
        // ...
    }
]
```

I've fixed the issue simply by removing the [environment override](https://github.com/GoogleContainerTools/skaffold/blob/main/pkg/skaffold/verify/docker/verify.go#L220) in sources and it works fine now, but I'm not sure if there are any reasons behind this behaviour, so I didn't open a PR.

### Expected behavior

Verify container has container's default environment variables

### Actual behavior

Verify container has host's environment variables

### Information

- Skaffold version: v2.0.0-beta3-28-gb5ce32043
- Operating system: MacOS 12.4
- Installed via: skaffold.dev or local build
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v3alpha1
kind: Config
build:
  artifacts:
    - image: test
      docker:
        dockerfile: Dockerfile
verify:
  - name: test
    container:
      name: test
      image: test
```

### Steps to reproduce the behavior

1. [a clonable repository with the sample skaffold project](https://github.com/quozd/skaffold-env-bug-demo)
2. `skaffold build -q > build_artifacts.json`
3. `skaffold verify -a build_artifacts.json`

Logs:
```
DEBU[0000] skaffold API not starting as it's not requested  subtask=-1 task=DevLoop
INFO[0000] Skaffold &{Version:v2.0.0-beta3-28-gb5ce32043 ConfigVersion:skaffold/v3alpha1 GitVersion: GitCommit:b5ce32043047b91205b398d9a1b4b539d4c673c4 BuildDate:2022-10-13T21:31:58Z GoVersion:go1.17 Compiler:gc Platform:darwin/arm64 User:}  subtask=-1 task=DevLoop
DEBU[0000] Skipping update check for pre-release version  subtask=-1 task=DevLoop
INFO[0000] Loaded Skaffold defaults from "/Users/<redacted>/.skaffold/config"  subtask=-1 task=DevLoop
DEBU[0000] parsed 1 configs from configuration file /Users/<redacted>/skaffold-env-bug-demo/skaffold.yaml  subtask=-1 task=DevLoop
DEBU[0000] Defaulting build type to local build          subtask=-1 task=DevLoop
DEBU[0000] Defaulting deploy type to kubectl             subtask=-1 task=DevLoop
INFO[0000] map entry found when executing locate for &{test . <nil> {0x14000bb0840 <nil> <nil> <nil> <nil> <nil> <nil>} [] {[] []} []} of type *latest.Artifact and pointer: 1374391879360  subtask=-1 task=DevLoop
INFO[0000] Using kubectl context: docker-desktop         subtask=-1 task=DevLoop
DEBU[0000] getting client config for kubeContext: `docker-desktop`  subtask=-1 task=DevLoop
DEBU[0000] getting client config for kubeContext: `docker-desktop`  subtask=-1 task=DevLoop
DEBU[0000] Running command: [minikube version --output=json]  subtask=-1 task=DevLoop
DEBU[0000] setting Docker user agent to skaffold-v2.0.0-beta3-28-gb5ce32043  subtask=-1 task=DevLoop
INFO[0000] manifests hydration will take place in /Users/<redacted>/skaffold-env-bug-demo/.kpt-pipeline  subtask=-1 task=DevLoop
DEBU[0000] CLI platforms provided: ""                    subtask=-1 task=DevLoop
DEBU[0000] platform detection from active kubernetes cluster is not enabled  subtask=-1 task=DevLoop
DEBU[0000] platforms selected for artifact "test": ""    subtask=-1 task=DevLoop
DEBU[0000] Using builder: local                          subtask=-1 task=DevLoop
DEBU[0000] push value not present in NewBuilder, defaulting to false because cluster.PushImages is false  subtask=-1 task=DevLoop
INFO[0000] build concurrency first set to 1 parsed from *local.Builder[0]  subtask=-1 task=DevLoop
INFO[0000] final build concurrency value is 1            subtask=-1 task=DevLoop
Tags used in verification:
 - test -> test:137d99c0f568bc460f8bc8b4ef662a268df0df1b69e561b4ba12f27dd25de577
DEBU[0000] push value not present in isImageLocal(), defaulting to false because cluster.PushImages is false  subtask=-1 task=DevLoop
DEBU[0000] Local images can't be referenced by digest.
They are tagged and referenced by a unique, local only, tag instead.
See https://skaffold.dev/docs/pipeline-stages/taggers/#how-tagging-works  subtask=-1 task=Verify
DEBU[0000] container test already present in local daemon: using test-2528b030 instead  subtask=test task=Verify
DEBU[0000] Running command: [tput colors]                subtask=-1 task=DevLoop
DEBU[0000] Command output: [256
]                        subtask=-1 task=DevLoop
1 error(s) occurred:
* creating container in local docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "go": executable file not found in $PATH: unknown
```

### PR DESCRIPTION
**Pull request by aaron-prindle**
fixes #7945

### CODE REVIEW

File: cmd/skaffold/app/cmd/flags.go [hunk 1]
```diff
@@ -543,6 +543,14 @@ var flagRegistry = []Flag{
 		FlagAddMethod: "DurationVar",
 		DefinedOn:     []string{"deploy", "dev", "run", "debug"},
 	},
+	{
+		Name:          "verify-env-file",
```
Comment (renzodavid9) [inline cmd/skaffold/app/cmd/flags.go:547]: Does it make sense to have a more generic name for this? Maybe `--env-file`? Just thinking maybe is better to have a single name for verify and custom actions (https://github.com/GoogleContainerTools/skaffold/issues/8560)

File: pkg/skaffold/verify/util/logfile.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// TimeFormat is used to name log files generated by deploy step
+const TimeFormat = "2006-01-02_15-04-05"
+
+type Muted interface {
+	MuteStatusCheck() bool
+	MuteDeploy() bool
+}
+
+// WithLogFile returns a file to write the deploy output to, and a function to be executed after the deploy step is complete.
+func WithLogFile(filename string, out io.Writer, muted Muted) (io.Writer, func(), error) {
```
Comment (renzodavid9) [inline pkg/skaffold/verify/util/logfile.go:36]: I think we are not using this function anywhere (except from the tests), is this something for an upcoming change? Or is ok to remove it?

File: pkg/skaffold/verify/util/namespaces.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	kubectx "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/kubernetes/context"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/util"
+)
+
+// GetAllPodNamespaces lists the namespaces that should be watched.
+// + The namespace passed on the command line
+// + Current kube context's namespace
+// + Namespaces referenced in Helm releases
+func GetAllPodNamespaces(configNamespace string, pipelines []latest.Pipeline) ([]string, error) {
```
Comment (renzodavid9) [inline pkg/skaffold/verify/util/namespaces.go:32]: Same here

Comment (aaron-prindle) [reply pkg/skaffold/verify/util/namespaces.go:32]: removed now

File: pkg/skaffold/verify/util/logfile.go (hunk 2 above)
Comment (aaron-prindle) [reply pkg/skaffold/verify/util/logfile.go:36]: removed now

File: cmd/skaffold/app/cmd/flags.go (hunk 1 above)
Comment (aaron-prindle) [reply cmd/skaffold/app/cmd/flags.go:547]: Good point!  Done

**Review by renzodavid9 — APPROVED** [review]

**Pull request merged**